Automattic/jetpack
# Search: Try splitting helper classes into separate package

#### Changes proposed in this Pull Request:
This is a proof-of-concept PR.

In preparation for the standalone Search plugin, this PR migrates Jetpack Search helper classes into a standalone **package** for use within Jetpack and Search (WIP) plugins.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* On your site with Search subscription, ensure that the following products work as expected:
	* (Classic) Search
	* Instant Search (using the modal overlay)
	* Customizer integration
	* Customberg (`/wp-admin/admin.php?page=jetpack-search-configure`).
